src/componentManifest.ts

```diff
@@ -103,8 +89,13 @@ export function makeDeviceManifest({
       const lang: string | undefined = file.translationLanguage;
 
       if (lang) {
-        if (manifest.i18n[lang] === undefined) manifest.i18n[lang] = {};
-        manifest.i18n[lang].resources = normalizeToPOSIX(file.relative);
+        if (locales[lang] === undefined) locales[lang] = {};
+        locales[lang].resources = normalizeToPOSIX(file.relative);
+      }
+
+      const isEntryPoint: boolean | undefined = file.isEntryPoint;
+      if (isEntryPoint) {
+        entryPoint = file.relative;
```

Comment:
I think we should enforce one entry-point per component, and error out if a second entry-point file is encountered in the stream. Same for companion and settings.